api/src/main/java/com/nicknathanjustin/streamercontracts/votes/VoteServiceImpl.java

```diff
@@ -0,0 +1,157 @@
+        }
+    }
+
+    @Override
+    public boolean isVotingComplete(@NonNull final ContractModel contractModel) {
+        validateContract(contractModel);
+
+        final UUID contractId = contractModel.getId();
+        final Optional<VoteModel> optionalProposerVote = voteModelRepository.findByContractIdAndVoterId(contractId, contractModel.getProposer().getId());
+        final Optional<VoteModel> optionalStreamerVote = voteModelRepository.findByContractIdAndVoterId(contractId, contractModel.getStreamer().getId());
+        final Timestamp now = new Timestamp(System.currentTimeMillis());
+        final boolean proposerAndStreamerHaveVoted = optionalProposerVote.isPresent() && optionalStreamerVote.isPresent();
+        final boolean contractHasExpired = now.after(contractModel.getExpiresAt());
+        return proposerAndStreamerHaveVoted ||
+               proposerMarkedContractCompleted(contractModel) ||
+               streamerMarkedContractFailed(contractModel) ||
+               contractHasExpired;
+    }
+
+    @Override
+    public VoteOutcome getVoteOutcome(@NonNull final ContractModel contractModel) {
+        if (!isVotingComplete(contractModel)) {
+            throw new IllegalArgumentException("Attempted to check if contract was completed before voting was finished. ContractId: " + contractModel.getId());
+        }
+
+        VoteOutcome voteOutcome = VoteOutcome.DISPUTE;
+        final UUID contractId = contractModel.getId();
+        final Optional<VoteModel> optionalProposerVote = voteModelRepository.findByContractIdAndVoterId(contractId, contractModel.getProposer().getId());
+        final Optional<VoteModel> optionalStreamerVote = voteModelRepository.findByContractIdAndVoterId(contractId, contractModel.getStreamer().getId());
+        if (proposerMarkedContractCompleted(contractModel)) {
+            log.info("Proposer has marked contract: {} as completed", contractModel.getId());
+            voteOutcome = VoteOutcome.COMPLETED;
+        } else if (streamerMarkedContractFailed(contractModel)) {
+            log.info("Streamer has marked contract: {} as failed", contractModel.getId());
+            voteOutcome = VoteOutcome.FAILED;
+        } else if(!optionalProposerVote.isPresent() && !optionalStreamerVote.isPresent()) {
+            log.info("Neither Streamer nor Proposer voted on contract: {}. contractModel.isAccepted(): {}",
+                    contractModel.getId(),
+                    contractModel.isAccepted());
+            voteOutcome = contractModel.isAccepted() ? VoteOutcome.COMPLETED : VoteOutcome.FAILED;
+        } else if(!optionalStreamerVote.isPresent()) {
+            log.info("Streamer did not vote on contract: {}. proposerVote.isViewerFlaggedComplete(): {}",
+                    contractModel.getId(),
+                    optionalProposerVote.get().isViewerFlaggedComplete());
+            voteOutcome = optionalProposerVote.get().isViewerFlaggedComplete() ? VoteOutcome.COMPLETED : VoteOutcome.FAILED;
```

Comment:
Will this logic (line 75) ever be used? If the viewer has flagged as complete, we are going to go into the first if statement, correct?

Reply:
You're right! Good eyes finding dead-code. Reworked to remove dead code.